app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -12,12 +12,35 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
+        create_rhel_eight_substitutions
```

Comment:
could this be renamed to something like `create_substitutions`? If the release-specific magic could live in its own class where we just add special cases as needed, that would be good.

Reply:
yes, I was thinking of a better way to structure this, I was thinking of a helper module with the methods that could be included here, but I'm not sure I can change the initializer easily. 

Maybe we could call RHEL8 and higher something? Like `modern_rhel` or something like that, so this would be `create_modern_rhel_substiutions`.

Reply:
I think the term 'substitutable' or 'non_substitutable' would be better, and a method named 'extract_hardcoded_substitutions' or something like that would be just as clear?